src/Agent.Worker/Release/ReleaseJobExtension.cs

```diff
@@ -221,25 +228,37 @@ public ReleaseJobExtension()
         private void CleanUpArtifactsFolder(IExecutionContext executionContext, string artifactsWorkingFolder)
         {
             Trace.Entering();
-            executionContext.Output(StringUtil.Loc("RMCleaningArtifactsDirectory", artifactsWorkingFolder));
-            try
-            {
-                if (Directory.Exists(artifactsWorkingFolder))
-                {
-                    IOUtil.DeleteDirectory(artifactsWorkingFolder, executionContext.CancellationToken);
-                }
 
-                Directory.CreateDirectory(artifactsWorkingFolder);
-            }
-            catch (Exception ex)
+            RetryExecutor retryExecutor = new RetryExecutor();
+            retryExecutor.ShouldRetryAction = (ex) =>
             {
+                executionContext.Output(StringUtil.Loc("FailedCleaningupRMArtifactDirectory", artifactsWorkingFolder));
```

Comment:
Good to Output the exception also (assuming Trace.Error will not log it to execution output)?
Also good to log that we are retrying the cleanup? Essentially - you can do:
Retrying the cleanup of Release artifacts directory, exception: {0}